gazelle/parser/ident.go

```diff
@@ -0,0 +1,207 @@
+		} else {
+			return &Function{Value: str}
+		}
+	}
+
+	// Otherwise, create an <ident-token> with its value set to string and return it.
+	return &Ident{Value: str}
+}
+
+func (s *Scanner) scanURL() Token {
+	// Note: This algorithm assumes that the initial "url(" has already been
+	// consumed. This algorithm also assumes that it’s being called to consume an
+	// "unquoted" value, like url(foo). A quoted value, like url("foo"), is
+	// parsed as a <function-token>. Consume an ident-like token automatically
+	// handles this distinction; this algorithm shouldn’t be called directly
+	// otherwise.
+
+	// Initially create a <url-token> with its value set to the empty string.
+	buf := strings.Builder{}
+
+	// Consume as much whitespace as possible.
+	for ch := s.peek(); isWhitespace(ch); ch = s.peek() {
+		s.read()
+	}
+
+	/* test code to print out all the runes that haven't been processed DO NOT SUBMIt
+	for {
+		if ch := s.read(); ch == eof {
+			return &URL{Value: buf.String()}
+		} else {
+			buf.WriteRune(ch)
+		}
+	}
+	/**/
+
+	// Repeatedly consume the next input code point from the stream:
+	for {
+		if ch := s.read(); ch == ')' {
+			// U+0029 RIGHT PARENTHESIS ())
+			// Return the <url-token>.
+			return &URL{Value: buf.String()}
+		} else if ch == eof {
+			// EOF
+			// This is a parse error. Return the <url-token>.
+			return &URL{Value: buf.String()}
+		} else if isWhitespace(ch) {
+			// whitespace
+			// Consume as much whitespace as possible.
+			for ch := s.peek(); isWhitespace(ch); ch = s.peek() {
+				s.read()
+			}
+
+			// If the next input code point is U+0029 RIGHT PARENTHESIS ()) or EOF, consume it and return the <url-token>
+			if ch := s.read(); ch == ')' {
+				return &URL{Value: buf.String()}
+			} else if ch == eof {
+				// (if EOF was encountered, this is a parse error);
+				// We choose to return a BadURL here.
+				return &BadURL{Value: "firstone"}
+				//return &BadURL{Value: s.scanBadURL()}
+			} else {
+				// otherwise, consume the remnants of a bad url, create a <bad-url-token>, and return it.
+				return &BadURL{Value: "otherone"}
+				//return &BadURL{Value: s.scanBadURL()}
+			}
+		} else if ch == '"' || ch == '\'' || ch == '(' || isNonASCIICodePoint(ch) {
+			// U+0022 QUOTATION MARK (")
+			// U+0027 APOSTROPHE (')
+			// U+0028 LEFT PARENTHESIS (()
+			// non-printable code point
+			// This is a parse error. Consume the remnants of a bad url, create a <bad-url-token>, and return it.
+			return &BadURL{Value: "thisone"}
+			//return &BadURL{Value: s.scanBadURL()}
+		} else if ch == '\\' {
+			// U+005C REVERSE SOLIDUS (\)
+			// If the stream starts with a valid escape, consume an escaped code point and append the returned code point to the <url-token>’s value.
+			buf.WriteString(s.scanEscape())
+		} else {
+			buf.WriteRune(ch)
+		}
+	}
+
+	// anything else
+	// Append the current input code point to the <url-token>’s value.
+	return &URL{Value: ""}
+}
+
+func (s *Scanner) scanBadURL() string {
```

Comment:
Is it standard to not camelCase acronyms in Go?

Reply:
Correct. Here is the link to the [initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) section of the code review comments doc (which is based on the golang styleguide). Interestingly, if you don't put "URL" or "ID" in all caps it causes a linter error. :man_shrugging: